### PR TITLE
Utilized option attributes

### PR DESCRIPTION
**Pull request by dorfire**

### CODE REVIEW

File: src/BuildScriptGeneratorCli/Commands/BuildCommand.cs [hunk 1]
```diff
@@ -38,6 +38,7 @@ internal class BuildCommand : CommandBase
         };
 
         [Argument(0, Description = "The source directory.")]
+        [DirectoryExists]
```
Comment (kichalla) [inline src/BuildScriptGeneratorCli/Commands/BuildCommand.cs]: The question is whether this works for the cases like `.`, `c:\foo\..\dir1` etc.

Comment (dorfire) [reply src/BuildScriptGeneratorCli/Commands/BuildCommand.cs]: Yeah, it works with relative paths as well.

Comment (kichalla) [reply src/BuildScriptGeneratorCli/Commands/BuildCommand.cs]: Currently we support doing `oryx build` without supplying any value - in which case current directory is taken. This change will break it?

Comment (dorfire) [reply src/BuildScriptGeneratorCli/Commands/BuildCommand.cs]: it'll probably break it - i'm surprised we don't have a test for that - adding one.

Comment (kichalla) [reply src/BuildScriptGeneratorCli/Commands/BuildCommand.cs]: I think we do have integration tests for it

File: tests/Oryx.BuildImage.Tests/OryxCommandTest.cs [hunk 2]
```diff
@@ -6,14 +6,36 @@
...
+                CommandToExecuteOnRun = "oryx",
+                CommandArguments = new[] { "build" },
+                WorkingDirectory = "/tmp"
+            });
+
+            // Assert
+            RunAsserts(
+                () =>
+                {
+                    Assert.Contains("Error: Could not detect", result.StdErr);
```
Comment (kichalla) [inline tests/Oryx.BuildImage.Tests/OryxCommandTest.cs]: Instead of an error case, we could just build a sample app of any platform and test for sucess

File: src/BuildScriptGeneratorCli/Commands/BuildCommand.cs (hunk 1 above)
Comment (kichalla) [reply src/BuildScriptGeneratorCli/Commands/BuildCommand.cs]: Revert this change then?

File: src/BuildScriptGeneratorCli/Commands/BuildScriptCommand.cs [hunk 3]
```diff
@@ -19,6 +19,7 @@ internal class BuildScriptCommand : CommandBase
         public const string Name = "build-script";
 
         [Argument(0, Description = "The source directory.")]
+        [DirectoryExists]
```
Comment (kichalla) [inline src/BuildScriptGeneratorCli/Commands/BuildScriptCommand.cs]: Same here. Revert this change.
